src/app/pages/question/question.component.ts

```diff
@@ -1,36 +1,92 @@
 import { Component, Input, OnInit } from '@angular/core';
 import { UIKitModule } from '../../ui-kit/ui-kit.module';
 import { ActivatedRoute } from '@angular/router';
-import { QUIZ_DATA } from '../../test-data/quiz';
 import { CommonModule } from '@angular/common';
+import { QuestionModel } from '../../services/model/question.model';
+import { StoreService } from '../../services/store.service';
+import {
+  BehaviorSubject,
+  combineLatest,
+  map,
+  Observable,
+  take,
+} from 'rxjs';
+import { QUESTIONS_SIZE } from '../../utils/constants';
 
 @Component({
   standalone: true,
   selector: 'app-quiz',
   templateUrl: './question.component.html',
   imports: [ UIKitModule, CommonModule ],
 })
-export class QuizComponent implements OnInit {
+export class QuestionComponent implements OnInit {
   @Input() quiz: string = '';
-  quiz_data = QUIZ_DATA;
-  currentIndex: number = 0;
   
-  constructor(private route: ActivatedRoute) {}
+  questions$!: Observable<QuestionModel[]>;
+  isLoading$!: Observable<boolean>;
+  currentIndex$ = new BehaviorSubject<number>(0);
+  currentQuestion$!: Observable<QuestionModel>;
+  answers: string[] = new Array<string>(QUESTIONS_SIZE);
+
+  constructor(
+    private route: ActivatedRoute,
+    private storeService: StoreService,
+  ) {}
 
   nextQuestion(): void {
-    if (this.currentIndex < this.quiz_data.length - 1) {
-      this.currentIndex++;
+    const currentIndex = this.currentIndex$.value;
+    if (currentIndex < QUESTIONS_SIZE - 1) {
+      this.currentIndex$.next(currentIndex + 1);
     }
   }
 
   prevQuestion(): void {
-    if (this.currentIndex > 0) {
-      this.currentIndex--;
+    const currentIndex = this.currentIndex$.value;
+    if (currentIndex > 0) {
+      this.currentIndex$.next(currentIndex - 1);
     }
   }
 
+  getSelectedAnswer(value: string): void {
+    this.answers[this.currentIndex$.value] = value;
+  }
+
+  get options(): Observable<string[]> {
+    return this.currentQuestion$.pipe(
+      take(1),
+      map((item) => {
+        return [ ...item.incorrect_answers, item.correct_answer ]
+        .sort()
+        .reverse();
+      }),
+    );
+  }
+
+  get id(): number {
+    return this.currentIndex$.value + 1;
+  }
+
   ngOnInit(): void {
     // eslint-disable-next-line dot-notation
-    this.quiz = this.route.snapshot.params['name'];
+    this.quiz = this.route.snapshot.params['id'];
```

Comment:
You can use here params.id and remove eslint disable

Reply:
If I use this `this.quiz = this.route.snapshot.params.id;`, I get this error
![image](https://github.com/user-attachments/assets/0615d4d6-449b-4c1e-9877-083fcaf7fdd2)
So, I decided to use with ['id]. 
If you have better solution of this, I'm waiting for your comment